classes/submission/Submission.inc.php

```diff
@@ -337,14 +374,20 @@ function getAuthorEmails() {
 	/**
 	 * Get all authors of this submission.
 	 * @param $onlyIncludeInBrowse boolean whether to limit to include_in_browse authors.
+	 * @param $version int
 	 * @return array Authors
 	 */
-	function getAuthors($onlyIncludeInBrowse = false) {
+	function getAuthors($onlyIncludeInBrowse = false, $version = null) {
 		$authorDao = DAORegistry::getDAO('AuthorDAO');
-		if (!$onlyIncludeInBrowse)
-			return $authorDao->getBySubmissionId($this->getId());
-		else
-			return $authorDao->getBySubmissionId($this->getId(), false, true);
+		if ($version == null) {
+			$version = $this->getCurrentVersionId() ? $this->getCurrentVersionId() : 1; // if a new submission is submitted, version is 1
+		}
+
+		if (!$onlyIncludeInBrowse) {
+			return $authorDao->getBySubmissionId($this->getId(), false, false, $version);
+		} else {
+			return $authorDao->getBySubmissionId($this->getId(), false, true, $version);
```

Comment:
These if... else... could maybe be reduces to only one call, giving $onlyIncludeInBrowse as parameter to the function getBySubmissionId, right?